app/models/comment.rb

```diff
@@ -46,14 +48,23 @@ class Comment < ActiveRecord::Base
     participation.unparticipate! if participation.present?
   end
 
-  def message
-    @message ||= Diaspora::MessageRenderer.new text
-  end
-
   def text= text
      self[:text] = text.to_s.strip #to_s if for nil, for whatever reason
   end
 
+  def people_allowed_to_be_mentioned
+    return @people_allowed_to_be_mentioned unless @people_allowed_to_be_mentioned.nil?
+    contacts = author.owner.try(:contact_people) if parent.public?
+
+    @people_allowed_to_be_mentioned = [*parent.participants, *contacts, parent.author].uniq
+  end
+
+  def subscribers
+    super.tap {|subscribers|
+      subscribers.concat(legitimate_mention_recipients) if parent.author.local?
```

Comment:
When the (mentioned) post author is local and the (mentioning) comment author is remote, it sends two notification mails. That is because you add the local mentioned people (which already received it) again to the subscribers for relaying, so they receive it again.